### PR TITLE
[WIP] #228 - New Rails Backend

### PR DESCRIPTION
**Pull request by tsimica**

### CODE REVIEW

**Comment by siefkenj** [comment]
Modify the rubocop rules in a suitable way for your code to pass. Also, it says `db:create` fails...Does the .travis.yml file need to be changed?

File: api/app/services/assignment_wage_chunk_create_service.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    def upsert_and_return_wage_chunks
+        @assignment.wage_chunks
+                   .upsert_all(@wage_chunks,
+                               returning: RETURNING_VALUES)
+    end
+
+    class WageChunkMissing < StandardError
+        def message
+            I18n.t('assignment_wage_chunk_service.wage_chunk_missing.message')
```
Comment (siefkenj) [inline api/app/services/assignment_wage_chunk_create_service.rb:50]: We're internationalizing our messages?

File: api/config/cable.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ development:
   adapter: async
 
 test:
-  adapter: async
+  adapter: test
 
 production:
   adapter: redis
-  url: redis://localhost:6379/1
+  url: <%= ENV.fetch("REDIS_URL") { "redis://localhost:6379/1" } %>
```
Comment (siefkenj) [inline api/config/cable.yml]: What's with the hard-coded URL and port?

File: api/config/routes.rb [hunk 3]
```diff
@@ -1,102 +1,91 @@
...
+
+                # Application
+                resources :applications, only: :create
+
+                # Assignments
+                resources :assignments, only: %i[show create] do
+                    resources :wage_chunks, controller: :assignment_wage_chunks, only: %i[index create]
+                    resources :offers, path: :active_offer, controller: :active_offers, only: :index do
+                        collection do
+                            post 'create'
```
Comment (siefkenj) [inline api/config/routes.rb:21]: Why is `'create'` a string while the rest are identifiers?

**Review by siefkenj — COMMENTED** [review]
Let's disable the failing tests for now, and we'll reactivate them in a new PR. (Theoretically, they should work if we prefix all the routes with `/admin`)

File: api/config/cable.yml (hunk 2 above)
Comment (tsimica) [reply api/config/cable.yml]: This was autogenerated. We don't need ActionCable for our use cases. I will remove it in a future commit.

File: api/config/routes.rb (hunk 3 above)
Comment (tsimica) [reply api/config/routes.rb:21]: `create` has special meaning in Rails and will give you a different route depending on whether it's a symbol or string. Defining `create` as a symbol will give you `POST /api/v1/offers/` vs. a string `POST api/v1/offers/create`

**Pull request merged**